vizro-core/tests/unit/vizro/models/_components/form/test_checklist.py

```diff
@@ -0,0 +1,144 @@
+
+        assert hasattr(checklist, "id")
+        assert checklist.type == "checklist"
+        assert checklist.options == expected
+        assert checklist.value is None
+        assert checklist.title is None
+        assert checklist.actions == []
+
+    @pytest.mark.parametrize("options", [1, "A", True, 1.0])
+    def test_create_checklist_invalid_options_type(self, options):
+        with pytest.raises(ValidationError, match="value is not a valid list"):
+            Checklist(options=options)
+
+    def test_create_checklist_invalid_options_dict(self):
+        with pytest.raises(
+            ValidationError,
+            match="Invalid argument `options` passed into Selector. Expected a dict with keys `label` and `value`.",
+        ):
+            Checklist(options=[{"hello": "A", "world": "A"}, {"hello": "B", "world": "B"}])
+
+    @pytest.mark.parametrize(
+        "options, test_value",
+        [
+            (["A", "B", "C"], ["A"]),
+            ([1, 2, 3], [1, 2]),
+            ([1.0, 2.0, 3.0], [1.0, 2.0]),
+            ([True, False], [False, True]),
+            ([{"label": "A", "value": "A"}, {"label": "B", "value": "B"}], ["A", "B"]),
+            ([True, 2.0, 1.0, "A", "B"], ["True", "A"]),
+        ],
+    )
+    def test_create_checklist_valid_value(self, options, test_value):
+        checklist = Checklist(options=options, value=test_value)
+
+        assert hasattr(checklist, "id")
+        assert checklist.type == "checklist"
+        assert checklist.value == test_value
+        assert checklist.title is None
+        assert checklist.actions == []
+
+    @pytest.mark.parametrize(
+        "test_value, options",
+        [
+            ([5], ["A", "B", "C"]),
+            ([5], [{"label": "A", "value": "A"}, {"label": "B", "value": "B"}]),
+            (["D"], ["A", "B", "C"]),
+            (["D"], [{"label": "A", "value": "A"}, {"label": "B", "value": "B"}]),
+            ([True], ["A", "B", "C"]),
+            ([True], [{"label": "A", "value": "A"}, {"label": "B", "value": "B"}]),
+        ],
+    )
+    def test_create_checklist_invalid_value_non_existing(self, test_value, options):
+        with pytest.raises(ValidationError, match="Please provide a valid value from `options`."):
+            Checklist(value=test_value, options=options)
+
+    def test_create_checklist_invalid_value_format(self):
+        with pytest.raises(ValidationError, match="value is not a valid list"):
+            Checklist(value="A", options=["A", "B", "C"])
+
+
+class TestChecklistBuild:
+    """Tests model build method."""
+
+    def test_checklist_build(self, expected_checklist):
+        checklist = Checklist(options=["A", "B", "C"], id="checklist_id", title="Title").build()
+
+        result = json.loads(json.dumps(checklist, cls=plotly.utils.PlotlyJSONEncoder))
+        expected = json.loads(json.dumps(expected_checklist, cls=plotly.utils.PlotlyJSONEncoder))
+        assert result == expected
```

Comment:
The fixture `expected_checklist` contains the option "ALL". Where does it come from if there is no multi=True set?

Reply:
The dcc.Checklist is by default a multi-select (so in that sense multi=True is always true). There is no multi option though. Similar to the dcc.RadioItem being a single-select by default. It's only the Dropdown where you can select between multi or single-select

Reply:
Yupp, I see now. Thanks.